compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs

```diff
@@ -164,10 +164,13 @@ decodeChoice LF1.TemplateChoice{..} =
     <*> mayDecode "templateChoiceRetType" templateChoiceRetType decodeType
     <*> mayDecode "templateChoiceUpdate" templateChoiceUpdate decodeExpr
 
+-- FixMe: https://github.com/digital-asset/daml/issues/2289
+--   NUMERIC and DECIMAL builtins should map to different internal builtin
 decodeBuiltinFunction :: MonadDecode m => LF1.BuiltinFunction -> m BuiltinExpr
 decodeBuiltinFunction = pure . \case
   LF1.BuiltinFunctionEQUAL_INT64 -> BEEqual BTInt64
   LF1.BuiltinFunctionEQUAL_DECIMAL -> BEEqual BTDecimal
+  LF1.BuiltinFunctionEQUAL_NUMERIC -> BEEqual BTDecimal
```

Comment:
This doesn't work since
```
EQUAL_DECIMAL : Decimal -> Decimal -> Bool
```
but
```
EQUAL_NUMERIC : forall (n : Nat). Numeric n -> Numeric n -> Bool
```
Similar issues below.

Reply:
fixed. Thanks for the help